GramAddict/plugins/data_analytics.py

```diff
@@ -0,0 +1,215 @@
+import json
+import matplotlib.pyplot as plt
+import os
+
+from datetime import timedelta, datetime
+from enum import Enum, unique
+from matplotlib import ticker
+from matplotlib.backends.backend_pdf import PdfPages
+from matplotlib.dates import DateFormatter
+
+from GramAddict.core.plugin_loader import Plugin
+
+A4_WIDTH_INCHES = 8.27
+A4_HEIGHT_INCHES = 11.69
+
+
+class DataAnalytics(Plugin):
+    """Generates a PDF analytics report of specified username session data"""
+
+    def __init__(self):
+        super().__init__()
+        self.description = (
+            "Generates a PDF analytics report of specified username session data"
+        )
+        self.arguments = [
+            {
+                "arg": "--analytics",
+                "nargs": 1,
+                "help": "generates a PDF analytics report of specified username session data",
+                "metavar": "username1",
+                "default": None,
+                "operation": True,
+            },
+        ]
+
+    def run(self, device, device_id, args, enabled, storage, sessions):
+        self.username = args.analytics[0]
+        sessions = self.load_sessions()
+        if not sessions:
+            return
+
+        filename = (
+            "report_"
+            + self.username
+            + "_"
+            + datetime.now().strftime("%Y-%m-%d")
+            + ".pdf"
+        )
+        with PdfPages(filename) as pdf:
+            sessions_week = self.filter_sessions(sessions, Period.LAST_WEEK)
+            sessions_month = self.filter_sessions(sessions, Period.LAST_MONTH)
+
+            self.plot_followers_growth(
+                sessions_week, pdf, self.username, Period.LAST_WEEK
+            )
+            self.plot_followers_growth(
+                sessions_month, pdf, self.username, Period.LAST_MONTH
+            )
+            self.plot_followers_growth(sessions, pdf, self.username, Period.ALL_TIME)
+
+            self.plot_duration_statistics(
+                sessions_week, pdf, self.username, Period.LAST_WEEK
+            )
+            self.plot_duration_statistics(
+                sessions_month, pdf, self.username, Period.LAST_MONTH
+            )
+            self.plot_duration_statistics(sessions, pdf, self.username, Period.ALL_TIME)
+
+        print("Report saved as " + filename)
+
+    def load_sessions(self):
+        path = self.username + "/sessions.json"
+        if os.path.exists(path):
+            with open(path) as json_file:
+                json_array = json.load(json_file)
+            return json_array
+        else:
+            print("No sessions.json file found for @" + self.username)
+            return None
+
+    def plot_followers_growth(self, sessions, pdf, username, period):
+        followers_count = [int(session["profile"]["followers"]) for session in sessions]
```

Comment:
Same here. If we make session to be a class with all attributes, it improves maintainability of the code.

`session.profile.followers` instead. Can you please create a task to improve this? We can merge it like this for now..